tests/testSpherePoint.py

```diff
@@ -0,0 +1,751 @@
+
+        for nonFinite in nonFiniteList:
+            self.assertIsInstance(nonFinite.isFinite(), bool)
+            self.assertFalse(nonFinite.isFinite())
+
+    def testGetItemError(self):
+        """Test if indexing correctly handles invalid input.
+        """
+        point = SpherePoint(afwGeom.Point3D(1.0, 1.0, 1.0))
+
+        with self.assertRaises(pexEx.OutOfRangeError):
+            point[2]
+        with self.assertRaises((pexEx.OutOfRangeError, OverflowError)):
+            point[-1]
+
+    def testGetItemValue(self):
+        """Test if indexing returns the expected value.
+        """
+        for lon, lat in self._dataset:
+            point = SpherePoint(lon, lat)
+            self.assertIsInstance(point[0], afwGeom.Angle)
+            self.assertIsInstance(point[1], afwGeom.Angle)
+            if not math.isnan(point.getLongitude().asRadians()):
+                self.assertEqual(point.getLongitude(), point[0])
+            else:
+                self.assertTrue(math.isnan(point[0].asRadians()))
+            if not math.isnan(point.getLatitude().asRadians()):
+                self.assertEqual(point.getLatitude(), point[1])
+            else:
+                self.assertTrue(math.isnan(point[1].asRadians()))
+
+    def testEquality(self):
+        """Test if tests for equality treat SpherePoints as values.
+        """
+        # (In)equality is determined by value, not identity.
+        # See DM-2347, DM-2465. These asserts are testing the
+        # functionality of `==` and `!=` and should not be changed.
+        for lon1, lat1 in self._dataset:
+            point1 = SpherePoint(lon1, lat1)
+            self.assertIsInstance(point1 == point1, bool)
+            self.assertIsInstance(point1 != point1, bool)
+            if point1.isFinite():
+                self.assertTrue(point1 == point1)
+                self.assertFalse(point1 != point1)
+
+                pointCopy = copy.deepcopy(point1)
+                self.assertIsNot(pointCopy, point1)
+                self.assertEqual(pointCopy, point1)
+                self.assertEqual(point1, pointCopy)
+                self.assertFalse(pointCopy != point1)
+                self.assertFalse(point1 != pointCopy)
+            else:
+                self.assertFalse(point1 == point1)
+                self.assertTrue(point1 != point1)
+
+            for lon2, lat2 in self._dataset:
+                if lon1 == lon2 and lat1 == lat2:
+                    continue
+                point2 = SpherePoint(lon2, lat2)
+                self.assertTrue(point2 != point1)
+                self.assertTrue(point1 != point2)
+                self.assertFalse(point2 == point1)
+                self.assertFalse(point1 == point2)
+
+        # Test for transitivity (may be assumed by algorithms).
+        for delta in [10.0**(0.1*x) for x in range(-150, -49, 5)]:
+            self._checkTransitive(delta*radians)
+
+    def _checkTransitive(self, delta):
```

Comment:
Consider making thisi public by deleting the leading underscore. It is our usual pattern.